### ISSUE TITLE
RAM Usage Issue

### ISSUE THREAD

**Issue by wahyuabrory** [creewick/uWidgets#38]
**Describe the bug**
Ram usage over time is insane

**Expected behavior**
At the start of using the widget on startup, it's pretty light. However, over time, the RAM usage increases. In my case, it eventually consumes 1GB of RAM.

**Screenshots**
![image](https://github.com/user-attachments/assets/47fcc4cb-6cc1-4627-bb29-6840ed7082e7)
Ram Usage

![image](https://github.com/user-attachments/assets/8815a1dc-b3c0-4d36-a1c7-51c5db037715)
my widget

**Desktop**
 - OS: [Windows 11 23H2]
 - Version: [0.5.4]

**Crash log**
```
Object reference not set to an instance of an object.
   at Clock.Views.World.OnSizeChanged(Object sender, SizeChangedEventArgs e)
   at Avalonia.Interactivity.EventRoute.RaiseEventImpl(RoutedEventArgs e)
   at Avalonia.Interactivity.EventRoute.RaiseEvent(Interactive source, RoutedEventArgs e)
   at Avalonia.Interactivity.Interactive.RaiseEvent(RoutedEventArgs e)
   at Avalonia.Controls.Control.OnPropertyChanged(AvaloniaPropertyChangedEventArgs change)
   at Avalonia.Controls.ContentControl.OnPropertyChanged(AvaloniaPropertyChangedEventArgs change)
   at Avalonia.Animation.Animatable.OnPropertyChangedCore(AvaloniaPropertyChangedEventArgs change)
   at Avalonia.AvaloniaObject.RaisePropertyChanged[T](AvaloniaProperty`1 property, Optional`1 oldValue, BindingValue`1 newValue, BindingPriority priority, Boolean isEffectiveValue)
   at Avalonia.AvaloniaObject.SetAndRaise[T](DirectPropertyBase`1 property, T& field, T value)
   at Avalonia.Layout.Layoutable.ArrangeCore(Rect finalRect)
   at Avalonia.Layout.Layoutable.Arrange(Rect rect)
   at Avalonia.Controls.Presenters.ContentPresenter.ArrangeOverrideImpl(Size finalSize, Vector offset)
   at Avalonia.Controls.Presenters.ContentPresenter.ArrangeOverride(Size finalSize)
   at Avalonia.Layout.Layoutable.ArrangeCore(Rect finalRect)
   at Avalonia.Layout.Layoutable.Arrange(Rect rect)
   at Avalonia.Layout.LayoutHelper.ArrangeChildInternal(Layoutable child, Size availableSize, Thickness padding)
   at Avalonia.Layout.LayoutHelper.ArrangeChild(Layoutable child, Size availableSize, Thickness padding, Thickness borderThickness)
   at Avalonia.Controls.Border.ArrangeOverride(Size finalSize)
   at Avalonia.Layout.Layoutable.ArrangeCore(Rect finalRect)
   at Avalonia.Layout.Layoutable.Arrange(Rect rect)
   at Avalonia.Controls.Grid.ArrangeOverride(Size arrangeSize)
   at Avalonia.Layout.Layoutable.ArrangeCore(Rect finalRect)
   at Avalonia.Layout.Layoutable.Arrange(Rect rect)
   at Avalonia.Controls.Presenters.ContentPresenter.ArrangeOverrideImpl(Size finalSize, Vector offset)
   at Avalonia.Controls.Presenters.ContentPresenter.ArrangeOverride(Size finalSize)
   at Avalonia.Layout.Layoutable.ArrangeCore(Rect finalRect)
   at Avalonia.Layout.Layoutable.Arrange(Rect rect)
   at Avalonia.Layout.LayoutHelper.ArrangeChildInternal(Layoutable child, Size availableSize, Thickness padding)
   at Avalonia.Layout.LayoutHelper.ArrangeChild(Layoutable child, Size availableSize, Thickness padding)
   at Avalonia.Controls.Primitives.VisualLayerManager.ArrangeOverride(Size finalSize)
   at Avalonia.Layout.Layoutable.ArrangeCore(Rect finalRect)
   at Avalonia.Layout.Layoutable.Arrange(Rect rect)
   at Avalonia.Layout.Layoutable.ArrangeOverride(Size finalSize)
   at Avalonia.Layout.Layoutable.ArrangeCore(Rect finalRect)
   at Avalonia.Layout.Layoutable.Arrange(Rect rect)
   at Avalonia.Layout.Layoutable.ArrangeOverride(Size finalSize)
   at Avalonia.Controls.WindowBase.ArrangeCore(Rect finalRect)
   at Avalonia.Layout.Layoutable.Arrange(Rect rect)
   at Avalonia.Layout.LayoutManager.Arrange(Layoutable control)
   at Avalonia.Layout.LayoutManager.ExecuteInitialLayoutPass()
   at Avalonia.Controls.Window.ShowCore(Window owner)
   at uWidgets.App.OnFrameworkInitializationCompleted()
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime(AppBuilder builder, String[] args, Action`1 lifetimeBuilder)
   at uWidgets.Program.Main(String[] args)
```

**Comment by creewick** [comment]
Woah, that's very high RAM usage
It supposed to be around ~100MB in general
I will look at this asap, looks like a memory leakage

**Comment by creewick** [comment]
![telegram-cloud-photo-size-4-5972184401007722975-y](https://github.com/user-attachments/assets/1c817e4a-3290-40aa-8e3d-250bb4a10177)

Memory leakage confirmed. I will need some more time to figure out the issue and fix it

**Comment by wahyuabrory** [comment]
take your time buddy

**Comment by creewick** [comment]
I found the cause of the memory leak. In the Monitor widget, I was using DoubleTransition to animate the circular bar. I will disable that until they fix that or any workaround will be available.

https://github.com/AvaloniaUI/Avalonia/issues/16973

The release will be here soon.

**Comment by creewick** [comment]
Fixed in 0.6.0
